extensions/basic_auth/README.md

```diff
@@ -57,24 +57,36 @@ message PluginConfig {
 
 // BasicAuth defines restriction rules based on three elements.
 message BasicAuth {
+  // A list of hosts that this basic auth rule applies on.
+  // Wildcard hosts are supported in the suffix or prefix form.
+  // For example:
+  //   Suffix domain wildcards: *.foo.com or *-bar.foo.com.
+  //   Prefix domain wildcards: foo.* or foo-*.
+  //
+  // If host is not provided, all hosts will be matched. A rule will
```

Comment:
Nit: if host is not provided, any host will match.

Reply:
Rule matches when at least one host in the list matches or the host list is empty.